packages/wrangler/src/pages/dev.ts

```diff
@@ -930,3 +830,203 @@ async function spawnProxyProcess({
 
 	return port;
 }
+
+/**
+ * Reconciles top-level & local development settings, with `pages dev`command line args taking
+ * precedence over `wrangler.toml` configuration
+ */
+function resolvePagesDevServerSettings(
+	config: Config,
+	args: PagesDevArguments
+) {
+	// resolve compatibility date
+	let compatibilityDate = args.compatibilityDate || config.compatibility_date;
+	if (!compatibilityDate) {
+		const currentDate = new Date().toISOString().substring(0, 10);
+		logger.warn(
+			`No compatibility_date was specified. Using today's date: ${currentDate}.\n` +
+				"Pass it in your terminal:\n" +
+				"```\n" +
+				`--compatibility-date=${currentDate}\n` +
+				"```\n" +
+				"See https://developers.cloudflare.com/workers/platform/compatibility-dates/ for more information."
+		);
+		compatibilityDate = currentDate;
+	}
+
+	return {
+		directory: args.directory ?? config.pages_build_output_dir,
+		compatibilityDate,
+		compatibilityFlags: args.compatibilityFlags ?? config.compatibility_flags,
+		ip: args.ip ?? config.dev.ip ?? DEFAULT_IP,
+		// because otherwise `unstable_dev` will default the port number to `0`
+		port: args.port ?? config.dev?.port ?? DEFAULT_LOCAL_PORT,
+		inspectorPort: args.inspectorPort ?? config.dev?.inspector_port,
+		localProtocol: args.localProtocol ?? config.dev?.local_protocol,
+	};
+}
+
+/**
+ * Parses the arguments specified by `pages dev` for environment bindings, and converts them
+ * into arrays of appropriate environment binding structures
+ */
+function getBindings(args: PagesDevArguments): Partial<
```

Comment:
I was thrown by the fact that this has the same function name as the one used by the normal dev command. This is really converting the command line args into override bindings that will be passed through to unstable_dev to be merged with those parsed from the wrangler.toml config, right?

Could we give this a bit more of an explicit name to call that out? E.g. `getBindingsFromArgs()` or `getOverrideBindings()` or something like that.

Reply:
renamed to `getBindingsFromArgs()`